component/src/main/java/org/wso2/extension/siddhi/execution/rdbms/QueryStreamProcessor.java

```diff
@@ -0,0 +1,253 @@
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * This extension function can be used to perform SQL retrieval queries on a wso2 datasource.
+ */
+@Extension(
+        name = "query",
+        namespace = "rdbms",
+        description = "The function can be used to perform SQL retrieval queries on a wso2 datasource. \n" +
+                "Note: This will only work within a carbon environment.",
+        parameters = {
+                @Parameter(
+                        name = "datasource.name",
+                        description = "The name of the wso2 datasource the query should be performed on",
+                        type = DataType.STRING
+                ),
+                @Parameter(
+                        name = "query",
+                        description = "The select query that needs to be performed",
+                        type = DataType.STRING
+                ),
+                @Parameter(
+                        name = "attribute.definition.list",
+                        description = "Comma separated list of `<AttributeName AttributeType>`. " +
+                                "It is expected that the SQL query will return the attributes in order, as in if one " +
+                                "attribute is defined here, the SQL query should return one column result set, " +
+                                "if more than one column is returned then the first column will be processed. The " +
+                                "Siddhi data types supported will be `STRING`, `INT`, `LONG`, `DOUBLE`, `FLOAT`, " +
+                                "`BOOL`. \n Mapping of the Siddhi data type to database data type can be done as " +
+                                "follows, \n" +
+                                "*Siddhi Datatype*->*Datasource Datatype*\n" +
+                                "`STRING`->`CHAR`,`VARCHAR`,`LONGVARCHAR`\n" +
+                                "`INT`&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;->`INTEGER`\n" +
+                                "`LONG`&nbsp;&nbsp;&nbsp;&nbsp;->`BIGINT`\n" +
+                                "`DOUBLE`->`DOUBLE`\n" +
+                                "`FLOAT`&nbsp;&nbsp;&nbsp;->`REAL`\n" +
+                                "`BOOL`&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;->`BIT`\n",
+                        type = DataType.STRING
+                )
+        },
+        returnAttributes = {
+                @ReturnAttribute(
+                        name = "custom",
+                        description = "The return attributes will be the ones defined in the parameter" +
+                                "`attribute.definition.list`.",
+                        type = DataType.STRING
```

Comment:
I think you can give a list here STRING, INT ...

Reply:
Fixed in 663e63b